modules/apps/frontend-js/frontend-js-spa-web/src/main/java/com/liferay/frontend/js/spa/web/internal/servlet/taglib/helper/SPAHelper.java

```diff
@@ -38,49 +33,56 @@
 
 import java.lang.reflect.Field;
 
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Objects;
 import java.util.ResourceBundle;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.CopyOnWriteArrayList;
-import java.util.function.Consumer;
-
-import org.osgi.framework.BundleContext;
-import org.osgi.framework.Filter;
-import org.osgi.framework.InvalidSyntaxException;
-import org.osgi.framework.ServiceReference;
-import org.osgi.framework.ServiceRegistration;
-import org.osgi.service.component.annotations.Activate;
+
 import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Deactivate;
-import org.osgi.service.component.annotations.Modified;
 import org.osgi.service.component.annotations.Reference;
-import org.osgi.util.tracker.ServiceTracker;
-import org.osgi.util.tracker.ServiceTrackerCustomizer;
 
 /**
  * @author Bruno Basto
  */
-@Component(
-	configurationPid = "com.liferay.frontend.js.spa.web.internal.configuration.SPAConfiguration",
-	service = SPAHelper.class
-)
+@Component(service = SPAHelper.class)
```

Comment:
We can simplfy and get rid of this entire class if we manage to handle the `getExcludedPathsJSONArray` and `getExcludedTargetPortletsJSONArray` cases in the dynamic include.

Because in the end, this class mostly wraps `spaConfiguration` as coming from the config APIs